xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-ui/src/main/resources/js/security/requiredrights/requiredRightsDialog.js

```diff
@@ -444,6 +446,128 @@ define('xwiki-requiredrights-dialog', [
 
             this.analysisResultsContainer.appendChild(panelGroup);
         }
+
+        /**
+         * @returns {number} the minimum width of the rights list in horizontal layout
+         */
+        #computeMinWidth()
+        {
+            // Temporarily set width to min-content to get the scrollWidth
+            const prevWidth = this.rightsList.style.width;
+            this.rightsList.style.width = 'min-content';
+            // Force reflow
+            this.rightsList.offsetWidth;
+            const minWidth = this.rightsList.scrollWidth;
+            this.rightsList.style.width = prevWidth;
+            return minWidth;
+        }
+
+        /**
+         * Update the class of the rights list to 'vertical' if the available width is less than the minimum width.
+         *
+         * @param minWidth the minimum width of the rights list in horizontal layout
+         */
+        #updateVerticalClass(minWidth)
+        {
+            const availableWidth = this.rightsList.clientWidth;
+
+            // Only update the class if the available width is not zero, i.e., the element is visible.
+            if (availableWidth === 0) {
+                return;
+            }
+
+            if (minWidth > availableWidth) {
+                this.rightsList.classList.add('vertical');
+            } else if (minWidth < availableWidth - 1) {
+                // Add some tolerance to avoid flickering.
+                this.rightsList.classList.remove('vertical');
+            }
+        }
+
+        /**
+         * Debounce a function to limit how often it can be called.
+         *
+         * @param fn the function to debounce
+         * @param delay the delay in milliseconds
+         * @returns {(function(...[*]): void)|*} a debounced version of the function
+         */
+        #debounce(fn, delay)
```

Comment:
I'm surprised we don't have an existing debounce method we can import from somewhere else.